monkey/infection_monkey/monkey.py

```diff
@@ -351,7 +352,8 @@ def try_exploiting(self, machine, exploiter):
         try:
             result = exploiter.exploit_host()
             if result:
-                self.successfully_exploited(machine, exploiter)
+                self.successfully_exploited(machine, exploiter) if exploiter.SHOULD_ADD_MACHINE_TO_EXPLOITED_SET else\
+                    self.successfully_exploited(machine, exploiter, False)
```

Comment:
Why not `self.successfully_exploited(machine, exploiter, exploiter.SHOULD_ADD_MACHINE_TO_EXPLOITED_SET)` ?